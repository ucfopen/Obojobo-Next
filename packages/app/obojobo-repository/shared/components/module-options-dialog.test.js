jest.mock('obojobo-document-engine/src/scripts/common/util/download-document')
jest.mock('../repository-utils')

import React from 'react'
import { create } from 'react-test-renderer'

import ModuleOptionsDialog from './module-options-dialog'

describe('ModuleOptionsDialog', () => {
	let defaultProps

	let mockRepositoryUtils
	let mockDownloadDocument

	const originalConfirm = window.confirm

	beforeEach(() => {
		jest.resetAllMocks()

		defaultProps = {
			draftId: 'mockDraftId',
			title: 'Mock Module Title',
			editor: 'mockEditorType'
		}

		mockRepositoryUtils = require('../repository-utils')
		mockRepositoryUtils.urlForEditor.mockReturnValue('/url/for/editor')
		mockDownloadDocument = require('obojobo-document-engine/src/scripts/common/util/download-document')
	})

	afterAll(() => {
		window.confirm = originalConfirm
	})

	test('renders correctly with standard expected props', () => {
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		expect(mockRepositoryUtils.urlForEditor).toHaveBeenCalledTimes(1)
		expect(mockRepositoryUtils.urlForEditor).toHaveBeenCalledWith('mockEditorType', 'mockDraftId')

		expect(component.toJSON()).toMatchSnapshot()
	})

	test('"Share" button calls showModulePermissions', () => {
		defaultProps.showModulePermissions = jest.fn()
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		//there are a few buttons on the page using this class
		component.root.findByProps({ id: 'moduleOptionsDialog-shareButton' }).props.onClick()

		expect(defaultProps.showModulePermissions).toHaveBeenCalledTimes(1)
		expect(defaultProps.showModulePermissions).toHaveBeenCalledWith(defaultProps)
	})

	test('"Manage Collections" button calls showModuleManageCollections', () => {
		defaultProps.showModuleManageCollections = jest.fn()
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		component.root
			.findByProps({ id: 'moduleOptionsDialog-manageCollectionsButton' })
			.props.onClick()

		expect(defaultProps.showModuleManageCollections).toHaveBeenCalledTimes(1)
		expect(defaultProps.showModuleManageCollections).toHaveBeenCalledWith(defaultProps)
	})

	test('"Version History" button calls showVersionHistory', () => {
		defaultProps.showVersionHistory = jest.fn()
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		component.root
			.findByProps({ id: 'moduleOptionsDialog-showVersionHistoryButton' })
			.props.onClick()

		expect(defaultProps.showVersionHistory).toHaveBeenCalledTimes(1)
		expect(defaultProps.showVersionHistory).toHaveBeenCalledWith(defaultProps)
	})

	test('"Download JSON" button calls downloadDocument with the correct arguments', () => {
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		//there are a few buttons on the page using this class
		component.root.findByProps({ id: 'moduleOptionsDialog-downloadJSONButton' }).props.onClick()

		expect(mockDownloadDocument.downloadDocument).toHaveBeenCalledTimes(1)
		expect(mockDownloadDocument.downloadDocument).toHaveBeenCalledWith('mockDraftId', 'json')
	})

	test('"Download XML" button calls downloadDocument with the correct arguments', () => {
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		//there are a few buttons on the page using this class
		component.root.findByProps({ id: 'moduleOptionsDialog-downloadXMLButton' }).props.onClick()

		expect(mockDownloadDocument.downloadDocument).toHaveBeenCalledTimes(1)
		expect(mockDownloadDocument.downloadDocument).toHaveBeenCalledWith('mockDraftId', 'xml')
	})

	test('"Delete" button brings up confirmation dialog, canceled', () => {
		defaultProps.deleteModule = jest.fn()
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		window.confirm = jest.fn()
		window.confirm.mockReturnValue(false)

		//there are a few buttons on the page using this class
		component.root.findByProps({ id: 'moduleOptionsDialog-deleteButton' }).props.onClick()

		expect(window.confirm).toHaveBeenCalledTimes(1)
		expect(window.confirm).toHaveBeenCalledWith('Delete "Mock Module Title" id: mockDraftId ?')

		expect(defaultProps.deleteModule).not.toHaveBeenCalled()
	})

	test('"Delete" button brings up confirmation dialog, confirmed', () => {
		defaultProps.deleteModule = jest.fn()
		const component = create(<ModuleOptionsDialog {...defaultProps} />)

		window.confirm = jest.fn()
		window.confirm.mockReturnValue(true)

		//there are a few buttons on the page using this class
		component.root.findByProps({ id: 'moduleOptionsDialog-deleteButton' }).props.onClick()

		expect(window.confirm).toHaveBeenCalledTimes(1)
		expect(window.confirm).toHaveBeenCalledWith('Delete "Mock Module Title" id: mockDraftId ?')

		expect(defaultProps.deleteModule).toHaveBeenCalledTimes(1)
		expect(defaultProps.deleteModule).toHaveBeenCalledWith('mockDraftId')
	})
})
