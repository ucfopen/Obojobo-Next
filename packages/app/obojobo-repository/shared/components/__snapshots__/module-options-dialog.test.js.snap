// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleOptionsDialog renders correctly with standard expected props 1`] = `
<div
  className="module-permissions-dialog"
>
  <div
    className="top-bar"
  >
    <div
      className="repository--module-icon--image"
    >
      <svg
        version="1.1"
        viewBox="0 0 130 150"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <pattern
            dangerouslySetInnerHTML={
              Object {
                "__html": "<image xlink:href=\\"/library/module-icon/mockDraftId\\" width=\\"150\\" height=\\"150\\" />",
              }
            }
            height="150"
            id="module-img-id-mockDraftId"
            patternUnits="userSpaceOnUse"
            width="150"
          />
        </defs>
        <path
          d="M56.29165124598851 4.999999999999999Q64.9519052838329 0 73.61215932167728 4.999999999999999L121.24355652982142 32.5Q129.9038105676658 37.5 129.9038105676658 47.5L129.9038105676658 102.5Q129.9038105676658 112.5 121.24355652982142 117.5L73.61215932167728 145Q64.9519052838329 150 56.29165124598851 145L8.660254037844387 117.5Q0 112.5 0 102.5L0 47.5Q0 37.5 8.660254037844387 32.5Z"
          fill="url(#module-img-id-mockDraftId)"
        />
      </svg>
    </div>
    <div
      className="module-title"
    >
      Mock Module Title
    </div>
    <button
      aria-label="Close"
      className="repository--button close-button"
      onClick={[Function]}
    >
      Ã—
    </button>
  </div>
  <div
    className="wrapper"
  >
    <h1
      className="title"
    >
      Module Options
    </h1>
    <div
      className="buttons-with-labels"
    >
      <a
        className="repository--button "
        href="/preview/mockDraftId"
        target="_blank"
      >
        Preview
      </a>
      <div
        className="label"
      >
        View with preview controls.
      </div>
      <a
        className="repository--button "
        href="/url/for/editor"
        target="_blank"
      >
        Edit
      </a>
      <div
        className="label"
      >
        Write, edit, and update.
      </div>
      <button
        className="repository--button "
        onClick={[Function]}
      >
        Share
      </button>
      <div
        className="label"
      >
        Add or remove collaborators.
      </div>
      <button
        className="repository--button "
        onClick={[Function]}
      >
        Version History
      </button>
      <div
        className="label"
      >
        View and restore previous versions.
      </div>
      <button
        className="repository--button "
        onClick={[Function]}
      >
        Manage Collections
      </button>
      <div
        className="label"
      >
        Add to or remove from private collections.
      </div>
      <button
        className="repository--button "
        onClick={[Function]}
      >
        Download JSON
      </button>
      <div
        className="label"
      >
        Download a copy in JSON format.
      </div>
      <button
        className="repository--button "
        onClick={[Function]}
      >
        Download XML
      </button>
      <div
        className="label"
      >
        Download a copy in XML format.
      </div>
      <a
        className="repository--button "
        href="/library/mockDraftId"
        target="_blank"
      >
        Public Page
      </a>
      <div
        className="label"
      >
        Visit this modules public page.
      </div>
      <button
        className="repository--button dangerous-button delete-button"
        onClick={[Function]}
      >
        Delete
      </button>
      <div
        className="label delete-label"
      >
        Say farewell.
      </div>
    </div>
    <button
      className="repository--button done-button secondary-button"
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</div>
`;
